repos/base-sel4/src/core/platform.cc

```diff
@@ -402,6 +393,28 @@ void Platform::_init_rom_modules()
 		     next <= last && element->id != SEL4_BOOTINFO_HEADER_PADDING;
 			 element = next)
 		{
+			if (element->id == SEL4_BOOTINFO_HEADER_X86_TSC_FREQ) {
+				struct tsc_freq {
+					uint32_t freq_mhz;
+				} __attribute__((packed));
+				if (sizeof(tsc_freq) + sizeof(*element) != element->len) {
+					error("unexpected tsc frequency format");
+					continue;
+				}
+
+				tsc_freq const * boot_freq = reinterpret_cast<tsc_freq const *>(reinterpret_cast<addr_t>(element) + sizeof(* element));
+
+				xml.node("hardware", [&] () {
+					xml.node("features", [&] () {
+						xml.attribute("svm", false);
+						xml.attribute("vmx", false);
+					});
+					xml.node("tsc", [&] () {
+						xml.attribute("freq_khz" , boot_freq->freq_mhz / 1000UL);
```

Comment:
IMO the calculation should be `kHz=1000*MHz`, right?

Reply:
You are correct. Must have been a brain lag on my part...